deep_gemm/include/deep_gemm/impls/sm90_fp8_gemm_1d2d.cuh

```diff
@@ -428,6 +430,21 @@ sm90_fp8_gemm_1d2d_impl(float* sfb, int* grouped_layout,
                 cute::tma_store_arrive();
             }
             __syncwarp();
+
+            if constexpr (kEnableOverlap) {
+                if (threadIdx.x < BLOCK_N / TMA_D_BLOCK_N) {
+                    cute::tma_store_wait<0>();
+                }
+
+                cg::coalesced_group group = cg::coalesced_threads();
+                group.sync();
+
+                __threadfence();
+
+                if (threadIdx.x == 0) {
+                    atomicAdd(signal + scheduler.current_group_idx * ceil_div(shape_m, BLOCK_M) + m_block_idx, 1);
```

Comment:
btw I am still a bit worried about this atomicAdd... (the code location issue in https://github.com/sgl-project/sglang/pull/9660#issuecomment-3230244330 is already solved and no problem)

Reply:
My naive understanding is the following (please correct me if I am wrong!):

> the thread initiating the TMA operation and the thread executing the write signal are not necessarily the same

`cutlass::arch::NamedBarrier(kNumMathThreads).sync();` ensures all math threads sync at this point, thus we know all tma_store_wait calls have finished at this point, thus when we do a `atom.add.release.gpu.global.s32` on thread 0, we already ensure tma_store_wait from thread 0,1,2,3,... all done.